File: src/Neuralm.Services/Neuralm.Services.TrainingRoomService/Neuralm.Services.TrainingRoomService.Messages/StartTrainingSessionResponse.cs
using Neuralm.Services.Common.Messages.Abstractions;
using Neuralm.Services.TrainingRoomService.Messages.Dtos;
using System;

namespace Neuralm.Services.TrainingRoomService.Messages
{
    /// <summary>
    /// Represents the <see cref="StartTrainingSessionResponse"/> class.
    /// </summary>
    public class StartTrainingSessionResponse : Response
    {
        /// <summary>
        /// Gets the training session.
        /// </summary>
        public TrainingSessionDto TrainingSession { get; set; }

        /// <summary>
        /// Initializes an instance of the <see cref="StartTrainingSessionResponse"/> class.
        /// </summary>
        /// <param name="requestId">The request id.</param>
        /// <param name="trainingSession">The training session.</param>
        /// <param name="message">The message.</param>
        /// <param name="success">The success flag.</param>
        public StartTrainingSessionResponse(Guid requestId, TrainingSessionDto trainingSession, string message, bool success = false) : base(requestId, message, success)
        {
            TrainingSession = trainingSession;
        }
        
        /// <summary>
        /// Initializes an instance of the <see cref="StartTrainingSessionResponse"/> class.
        /// SERIALIZATION CONSTRUCTOR.
        /// </summary>
        public StartTrainingSessionResponse()
        {
            
        }
    }
}

File: src/Neuralm.Presentation.Web/src/messages/responses/StartTrainingSessionResponse.ts
import Response from './Response';
import TrainingSession from '../../models/TrainingSession';

/**
 * Represents the StartTrainingSessionResponse class.
 */
export default class StartTrainingSessionResponse extends Response {
  public trainingSession: TrainingSession;

  /**
   * Initializes a new instance of the StartTrainingSessionResponse class.
   */
  constructor(id: string, requestId: string, dateTime: Date, message: string, success: boolean, trainingSession: TrainingSession) {
    super(id, requestId, dateTime, message, success);
    this.trainingSession = trainingSession;
  }
}
